src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,49 @@
+package duke;
+
+import duke.tasks.TaskManager;
+import duke.tasks.TaskType;
+import duke.command.*;
+
+public class Parser {
+
+
+    public Parser(){
+    }
+
+
+    // Reads the input of the line to determine the command and run it
+    public Command parseCommand(TaskManager taskList, String userInput){
+
+        // Get command from the userInput
+        final String[] splitLine = userInput.split(" ", 2);
+        final String command = splitLine[0];
+
+        // Remove command from userInput
+        final String arguments = userInput.replaceAll( RegexType.START_LINE_REGEX.getRegexType() + command + RegexType.SPACE_REGEX.getRegexType(), "");
+
+        // Check the command type then execute the commands
+        if(arguments.equals(InputType.LIST.getInputType())) {
+            return new ListCommand(taskList);
+        }
```

Comment:
Doing this kind of defeats the purpose of using enum, right? Instead of using enum purely as strings, I suggest parsing the user input into enum, then using the enum to return the appropriate commands. 

An additional suggestion is to use switch case, which is recommended for enum since you can do something like: 
```suggestion
        switch (inputType) {
        case LIST: 
            return new ListCommand(taskList);
```